Source/App.swift

```diff
@@ -1036,12 +1036,21 @@ class App
             }
             let filePath = sourceAssetFolder! + "/" + file
             var destFile:String
-            if (useRoot) {
+
+            if (type == .javascript || type == .nodejs) {
+                var subFolder = "/device_icons/"
+                if (file.fileNameOnly().contains("_control_icon")) {
+                    subFolder = "/control_icons/"
```

Comment:
This looks like application specific code. If you would develop a javascript app that has nothing do do with device icons or control icons this subfolder would not be appropriate. Instead you would configure the subfolder that you would like the icons to end up in. Let's chat on possible ways of doing that offline.